### PR TITLE
Add Threading to BareMetal

### PR DESCRIPTION
**Pull request by tgiphil**

### CODE REVIEW

File: Demos/Run-BareMetal.Starter.x86.bat
Comment (AnErrupTion) [inline Demos/Run-BareMetal.Starter.x86.bat:1]: I don't think we should be adding back a starter project in the solution. The template works well for that, why should we do that?

File: Demos/Run-BareMetal.TestWorld.x86.bat
Comment (AnErrupTion) [inline Demos/Run-BareMetal.TestWorld.x86.bat:1]: For TestWorld, what we could do instead is to have it as a single project: we remove the Mosa.BareMetal.TestWorld library (it's unlikely we're going to use it anyway), and only have the Mosa.BareMetal.TestWorld.x86 project instead.

Comment (tgiphil) [reply Demos/Run-BareMetal.TestWorld.x86.bat:1]: I will need it for x64 and ARM work.

File: Demos/Run-BareMetal.Starter.x86.bat
Comment (tgiphil) [reply Demos/Run-BareMetal.Starter.x86.bat:1]: Let's include it to let's people work on both the kernel and their app at the same time. We can revisit this later.

**Pull request merged**